src/plugins/clips-executive/clips/domain.clp

```diff
@@ -247,15 +247,31 @@
   "Remove an atomic precondition from its parent and clean up the precondition
    tree. If the parent is a disjunction with no other disjunct, simplify it to
    true by removing it recursively. If it is a negation, remove it recursively.
-   If it's a conjunction, only remove the conjunct."
-  (?precond-name)
-  (do-for-fact
-    ((?precond domain-atomic-precondition) (?parent domain-precondition))
-    (and (eq ?precond:name ?precond-name) (eq ?precond:part-of ?parent:name))
-    (if (or (eq ?parent:type disjunction) (eq ?parent:type negation)) then
-      (remove-precondition ?parent:name)
+   If it's a conjunction, only remove the conjunct.
+   If the top-most precondition is removed, replace it by a trivially true one
+   (empty conjunction)."
+  (?precond-fact)
+  (bind ?parent-facts (find-fact ((?parent domain-precondition))
+               (eq (fact-slot-value ?precond-fact part-of) ?parent:name)))
+  (if (= (length$ ?parent-facts) 1)
```

Comment:
Why not use `do-for-fact` and for the case that there is no parent, you can do:
```
(if (not (do-for-fact ...))
 then
  ; There is no parent
)
```

Reply:
Ah yea true, somehow completely forgot that :) fixed now.